pkg/stave/list.go

```diff
@@ -0,0 +1,427 @@
+			nsByName[it.groupName] = append(nsByName[it.groupName], it)
+		case targetGroupImport:
+			impByLabel[it.groupName] = append(impByLabel[it.groupName], it)
+			if it.groupMeta != "" {
+				impMetaByLabel[it.groupName] = it.groupMeta
+			}
+		}
+	}
+
+	slices.SortFunc(locals, compareTargetItems)
+
+	var localGroups []targetGroup
+	if len(locals) > 0 {
+		localGroups = append(localGroups, targetGroup{header: "", items: locals})
+	}
+
+	return targetSections{
+		local:      localGroups,
+		namespaces: buildGroups(nsByName, nil),
+		imports:    buildGroups(impByLabel, impMetaByLabel),
+	}
+}
+
+func writeTable(
+	out io.Writer,
+	headerStyle, subsectionStyle lipgloss.Style,
+	group targetGroup,
+	renderName func(name string, isDefault bool) string,
+	indent string,
+) {
+	if len(group.items) == 0 {
+		return
+	}
+
+	if group.header != "" {
+		_, _ = fmt.Fprintln(out)
+		subtitle := group.header
+		if group.meta != "" {
+			subtitle = fmt.Sprintf("%s (%s)", group.header, group.meta)
+		}
+		_, _ = fmt.Fprintln(out, subsectionStyle.Render(subtitle))
+	}
+
+	type row struct {
+		name      string
+		usage     string
+		synopsis  string
+		isDefault bool
+	}
+
+	rows := make([]row, 0, len(group.items)+1)
+	rows = append(rows, row{
+		name:     "NAME",
+		usage:    "USAGE",
+		synopsis: "SYNOPSIS",
+	})
+
+	for _, it := range group.items {
+		syn := strings.TrimSpace(it.synopsis)
+		if syn == "" {
+			syn = "-"
+		}
+		rows = append(rows, row{
+			name:      it.displayName,
+			usage:     it.usage,
+			synopsis:  syn,
+			isDefault: it.isDefault,
+		})
+	}
+
+	// Column widths (ANSI-aware via lipgloss.Width).
+	maxName, maxUsage, maxSyn := 0, 0, 0
+	for _, r := range rows {
+		maxName = max(maxName, lipgloss.Width(r.name))
+		maxUsage = max(maxUsage, lipgloss.Width(r.usage))
+		maxSyn = max(maxSyn, lipgloss.Width(r.synopsis))
+	}
+
+	pad := func(text string, width int) string {
+		if width <= 0 {
+			return text
+		}
+		textWidth := lipgloss.Width(text)
+		if textWidth >= width {
+			return text
+		}
+		return text + strings.Repeat(" ", width-textWidth)
+	}
+
+	// Print header.
+	h := rows[0]
+	headerLine := strings.Join([]string{
+		pad(h.name, maxName),
+		pad(h.usage, maxUsage),
+		pad(h.synopsis, maxSyn),
+	}, "  ")
+	_, _ = fmt.Fprintln(out, indent+headerStyle.Render(headerLine))
+
+	// Print rows.
+	for _, r := range rows[1:] {
+		name := renderName(r.name, r.isDefault)
+		line := strings.Join([]string{
+			pad(name, maxName),
+			pad(r.usage, maxUsage),
+			pad(r.synopsis, maxSyn),
+		}, "  ")
+		_, _ = fmt.Fprintln(out, indent+line)
+	}
+}
+
+func enableColorForList() bool {
+	// Use auto-detection (like stave --version), not opt-in env var.
+	// Respects NO_COLOR and TERM blacklist via st.ColorEnabled().
+	return st.ColorEnabled()
+}
+
+func lowerFirstTargetName(s string) string {
+	parts := strings.Split(s, ":")
+	for i := range parts {
+		parts[i] = lowerFirstWord(parts[i])
+	}
+	return strings.Join(parts, ":")
+}
```

Comment:
Lol, in this case the `lo` version is probably _considerably_ less readable than its spelled-out counterpart 😅 But just in case you wanted to see what the fully-`lo`-brained version looks like...:


```suggestion
func lowerFirstTargetName(s string) string {
	return strings.Join(lo.Map(strings.Split(s, ":"), func(s string, _ int) string {
		return lowerFirstWord(s)
	}), ":")
}
```